### PR TITLE
Restructure storage samples

### PR DESCRIPTION
**Pull request by JesseLovelace**
Moves storage samples to their own files and adds several samples

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/CreateBucketWithStorageClassAndLocation.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    Bucket bucket =
+        storage.create(
+            BucketInfo.newBuilder(bucketName)
+                // See here for possible values: http://g.co/cloud/storage/docs/storage-classes
+                .setStorageClass(StorageClass.COLDLINE)
+                // Possible values: http://g.co/cloud/storage/docs/bucket-locations#location-mr
+                .setLocation("asia")
+                .build());
+    System.out.println(
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/CreateBucketWithStorageClassAndLocation.java]: I think this line could be in 2 lines maybe?

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/MoveObject.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    System.out.println(
+        "Moved object "
+            + objectName
+            + " from bucket "
+            + sourceBucketName
+            + " to "
+            + copiedBlob.getBucket());
+  }
+}
+// [END storage_move_file]
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/MoveObject.java]: Add a newline.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/RotateObjectEncryptionKey.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    // String projectId = "your-project-id";
+
+    // The ID of your GCS bucket
+    // String bucketName = "your-unique-bucket-name";
+
+    // The ID of your GCS object
+    // String objectName = "your-object-name";
+
+    // The encryption key originally used to encrypt the object
+    // String oldEncryptionKey = "TIbv/fjexq+VmtXzAlc63J4z5kFmWJ6NdAPQulQBT7g="
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/RotateObjectEncryptionKey.java]: Explain that these are base64 encoded. Also could you add context of what kind of encryption key it needs to be?

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBucketSnippets.java [hunk 4]
```diff
@@ -97,15 +125,13 @@ public void testBucket() throws InterruptedException {
...
-    assertTrue(blobSet.contains(blob2));
-    assertTrue(blobSet.contains(blob3));
-    assertTrue(blobSet.contains(blob4));
+    /**
+     * Set<Blob> blobSet = Sets.newHashSet(bucketSnippets.listBlobs().iterateAll()); while
+     * (blobSet.size() < 4) { Thread.sleep(500); blobSet =
+     * Sets.newHashSet(bucketSnippets.listBlobs().iterateAll()); }
+     * assertTrue(blobSet.contains(blob1)); assertTrue(blobSet.contains(blob2));
+     * assertTrue(blobSet.contains(blob3)); assertTrue(blobSet.contains(blob4));*
+     */
```
Comment (frankyn) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBucketSnippets.java]: Is this expected?

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBucketSnippets.java [hunk 5]
```diff
@@ -133,61 +159,209 @@ public void testBucket() throws InterruptedException {
     assertTrue(bucketSnippets.deleteDefaultAcl());
     assertNull(bucketSnippets.getDefaultAcl());
     thrown.expect(StorageException.class);
-    assertTrue(bucketSnippets.delete());
+    // assertTrue(bucketSnippets.delete());
```
Comment (frankyn) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBucketSnippets.java]: Commented section.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java [hunk 6]
```diff
@@ -106,74 +120,232 @@ public void after() {
...
+    String snippetOutput = snippetOutputCapture.toString();
+    assertTrue(snippetOutput.contains(BLOB));
+    System.setOut(System.out);
+  }
+
+  @Test
+  public void testListObjectsWithPrefix() {
+    final ByteArrayOutputStream snippetOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(snippetOutputCapture));
+    storage.create(BlobInfo.newBuilder(BlobId.of(BUCKET, "/a/1.txt")).build());
```
Comment (frankyn) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java]: Recommend removing the forward slash for these names.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITStorageSnippets.java [hunk 7]
```diff
@@ -239,20 +149,14 @@ public void testGetBucketWithMetageneration() {
...
-  }
-
+  /**
+   * @Test public void testListBucketsWithSizeAndPrefix() throws InterruptedException { Page<Bucket>
+   * buckets = storageSnippets.listBucketsWithSizeAndPrefix(BUCKET); while
+   * (Iterators.size(buckets.iterateAll().iterator()) < 1) { Thread.sleep(500); buckets =
+   * storageSnippets.listBucketsWithSizeAndPrefix(BUCKET); } Iterator<Bucket> bucketIterator =
+   * buckets.iterateAll().iterator(); while (bucketIterator.hasNext()) {
+   * assertTrue(bucketIterator.next().getName().startsWith(BUCKET)); } }*
+   */
```
Comment (frankyn) [inline google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITStorageSnippets.java]: Should this be removed instead?

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java (hunk 6 above)
Comment (JesseLovelace) [reply google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java]: I used them to be consistent with samples in other languages, for example see the Go/Ruby/Python samples here: https://cloud.google.com/storage/docs/listing-objects#storage-list-objects-nodejs

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/DisableLifecycleManagement.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        // The ID of your GCP project
+        // String projectId = "your-project-id";
+
+        // The ID of your GCS bucket
+        // String bucketName = "your-unique-bucket-name";
+
+        Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+        Bucket bucket = storage.get(bucketName);
+
+        //TODO add once api is implemented
```
[the remark this answers is not in the document]
Comment (JesseLovelace) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/DisableLifecycleManagement.java]: It's not, see https://github.com/googleapis/java-storage/issues/20 and https://github.com/googleapis/java-storage/pull/28/files

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java (hunk 6 above)
Comment (frankyn) [reply google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java]: I updated Python and working through the others. I recommend staying away from '/a/1.txt' if possible as has caused user confusion.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/DisableLifecycleManagement.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (frankyn) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/DisableLifecycleManagement.java]: Understand, should this be removed from this PR and moved into Pr#28 instead?

[the remark this answers is not in the document]
Comment (frankyn) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/buckets/DisableLifecycleManagement.java]:
~~~ Updated my comment due to typos~~~

**Pull request merged**